src/managers/GuildManager.js

```diff
@@ -36,6 +43,45 @@ class GuildManager extends BaseManager {
    * @typedef {Guild|GuildChannel|GuildMember|GuildEmoji|Role|Snowflake|Invite} GuildResolvable
    */
 
+  /**
+   * Partial data for a Role.
+   * @typedef {Object} PartialRoleData
+   * @property {number} id The ID for this role, used to set channel overrides,
+   * this is a placeholder and will be replaced by the API after consumption
+   * @property {string} [name] The name of the role
+   * @property {ColorResolvable} [color] The color of the role, either a hex string or a base 10 number
+   * @property {boolean} [hoist] Whether or not the role should be hoisted
+   * @property {number} [position] The position of the role
+   * @property {PermissionResolvable|number} [permissions] The permissions of the role
+   * @property {boolean} [mentionable] Whether or not the role should be mentionable
+   */
+
+  /**
+   * Partial overwrite data.
+   * @typedef {Object} PartialOverwriteData
+   * @property {number|Snowflake} id The Role or User ID for this overwrite
+   * @property {string} [type] The type of this overwrite
+   * @property {PermissionResolvable} [allow] The permissions to allow
+   * @property {PermissionResolvable} [deny] The permissions to deny
+   */
+
+  /**
+   * Partial data for a Channel.
+   * @typedef {Object} PartialChannelData
+   * @property {number} [id] The ID for this channel, used to set its parent,
+   * this is a placeholder and will be replaced by the API after consumption
+   * @property {number} [parentID] The parent ID for this channel
+   * @property {string} [type] The type of the channel
+   * @property {string} [name] The name of the channel
```

Comment:
The name is not optional.
```json
{
    "code": 50035,
    "errors": {
        "channels": {
            "0": {
                "name": {
                    "_errors": [
                        {
                            "code": "BASE_TYPE_REQUIRED",
                            "message": "This field is required"
                        }
                    ]
                }
            }
        }
    },
    "message": "Invalid Form Body"
}
```

```suggestion
   * @property {string} name The name of the channel
```